From d7afaec82f38d31188f12659249b7641f1241bbb Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Thu, 03 Feb 2022 14:11:24 +0200
Subject: [PATCH] [SW-68803] security: add jtag control functions

Adding functions to enable/disable jtag. This is in preparation to
control debug interfaces in boot rom flow.

Change-Id: I43750f9e93c2295ef38822add89e61a6c662cf87
---

diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 3d6c0f8..5d9bebf 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -160,8 +160,24 @@
 void security_i2c_s_rr_hbw_config(__unused uint8_t spi_controller_en) {}
 void security_smi_rr_hbw_config(__unused uint8_t spi_controller_en) {}
 void security_jtag_rr_hbw_config(__unused uint8_t spi_controller_en) {}
-void security_disable_jtag_if(void) {}
 void security_handle_twi(void) {}
+
+void security_jtag_dap(uint8_t status)
+{
+	regmap_writel(status, mmD0_PSOC_CS_TRACE_SECURITY_CTRL);
+}
+
+void security_disable_jtag_if(void)
+{
+	security_jtag_dap(SECURITY_JTAG_DAP_DISABLE);
+	/* TODO: Add additional RR config to lock jtag */
+}
+
+void security_enable_jtag_if(void)
+{
+	security_jtag_dap(SECURITY_JTAG_DAP_ENABLE);
+	/* TODO: Add additional RR config to lock jtag */
+}
 
 /* TODO: get newly added key addresses from global specs once it is updated */
 #define mmD0_PSOC_SECURITY_PUBLIC_KEY_3_0		0xC494190
@@ -302,8 +318,6 @@
 	return !!(config_fuse & EFUSE_SYSCFG_ENABLE_VALIDATION_MASK);
 }
 
-void security_enable_jtag_if(void) {}
-void security_jtag_dap(__unused uint8_t status) {}
 void check_and_enable_4b_termination(void) {}
 void security_debug_if_shared(void) {}
 void security_pcie_rr_config(__unused uint8_t spi_en) {}
